packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -128,6 +143,31 @@ public ResponseEntity<Integer> uploadDocumentFromGithub(
 
 	}
 
+	@PutMapping(value = "/{id}/uploadDocumentFromDOI")
+	public ResponseEntity<Integer> uploadDocumentFromDOI(
+		@PathVariable("id") String id,
+		@RequestParam(name = "doi", required = true) String doi,
+		@RequestParam(name = "filename", required = true) String filename
+	) throws IOException, URISyntaxException {
+			CloseableHttpClient httpclient = HttpClients.custom()
+				.disableRedirectHandling()
+				.build();	
+
+			byte[] fileAsBytes = downloadService.getPDF("https://unpaywall.org/" + doi);
```

Comment:
Need to test when you get nothing.

Reply:
Do we want to write server tests for this (and others) too?

Reply:
ive added a null and empty check

Reply:
@dvince2 I can look into adding some server tests - maybe perhaps apart of another PR?